okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -344,6 +347,73 @@ static Status toGrpcStatus(ErrorCode code) {
     return ERROR_CODE_TO_STATUS.get(code);
   }
 
+  private Socket upgradeToTls(Socket socket, SSLSocketFactory sslSocketFactory) throws IOException {
```

Comment:
I don't think I'd be happy with a TODO. I feel that the approach taken here is duplicating code that is handled elsewhere, and using that pre-existing code is relatively easy. Is that not the case?